app/src/main/java/com/duckduckgo/app/appearance/AppearanceActivity.kt

```diff
@@ -82,6 +87,28 @@ class AppearanceActivity : DuckDuckGoActivity() {
             .show()
     }
 
+    private val experimentalUIToggleListener = CompoundButton.OnCheckedChangeListener { _, isChecked ->
+        viewModel.onExperimentalUIModeChanged(isChecked)
+
+        TextAlertDialogBuilder(this)
+            .setTitle(R.string.appearanceNightModeDialogTitle)
+            .setMessage(R.string.appearanceNightModeDialogMessage)
+            .setPositiveButton(R.string.appearanceNightModeDialogPrimaryCTA)
+            .setNegativeButton(R.string.appearanceNightModeDialogSecondaryCTA)
+            .addEventListener(
+                object : TextAlertDialogBuilder.EventListener() {
+                    override fun onPositiveButtonClicked() {
+                        FireActivity.triggerRestart(baseContext, false)
+                    }
+
+                    override fun onNegativeButtonClicked() {
+                        // no-op
+                    }
```

Comment:
Why would we ever not want to relaunch the app? Looks like the effects of continuing on without restarting are very mixed.

What if we added to the toggle's description that "changing this setting will relaunch the app" and skip the dialog, to make it more straightforward?

Reply:
Yeah why not, for this specific scenario we can keep it simple